cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/DataSetsModules.java

```diff
@@ -119,7 +119,7 @@ protected void configure() {
 
   @Override
   public Module getDistributedModules() {
-    return getDistributedModules(false);
+    return getDistributedModules(true);
```

Comment:
Passing true here binds `MetadataChangePublisher` to `KafkaMetadataChangePublisher`, whereas false binds `MetadataChangePublisher` to `NoOpMetadataChangePublisher`.

DefaultMetadataStore checks cConf at the beginning of every method to determine whether or not to publish a metadata update. Before this change, when publishing is enabled via cConf, `MetadataChangePublisher` is sometimes bound to `KafkaMetadataChangePublisher` and sometimes bound to `NoOpMetadataChangePublisher`. This results in particular types of metadata updates not being published since `NoOpMetadataChangePublisher` is used.